src/test/java/org/broadinstitute/hellbender/engine/dataflow/DataflowTestData.java

```diff
@@ -0,0 +1,215 @@
+package org.broadinstitute.hellbender.engine.dataflow;
+
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.common.collect.Lists;
+import com.google.genomics.v1.Read;
+import htsjdk.samtools.SAMRecord;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.FakeReferenceSource;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReadContextData;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReferenceShard;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.VariantShard;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.variant.SkeletonVariant;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public class DataflowTestData {
+    private final List<KV<Integer, Integer>> readStartLength;
+
+    private final List<GATKRead> reads;
+    private final List<KV<ReferenceShard, Iterable<GATKRead>>> kvRefShardiReads;
+    private final List<SimpleInterval> readIntervals;
+    private final List<SimpleInterval> allIntervals;
+    private final List<KV<ReferenceBases, Iterable<GATKRead>>> kvRefBasesiReads;
+    private final List<KV<VariantShard, GATKRead>> kvVariantShardRead;
+    private final List<Variant> variants;
+    private final List<KV<VariantShard, Variant>> kvVariantShardVariant;
+    private final List<KV<GATKRead, ReferenceBases>> kvReadsRefBases;
+    private final List<KV<GATKRead, Variant>> kvReadVariant;
+    private final List<KV<GATKRead, Iterable<Variant>>> kvReadiVariant;
+    private final List<KV<GATKRead, Iterable<Variant>>> kvReadiVariant2;
+    private final List<KV<GATKRead, ReadContextData>> kvReadContextData;
+
+
+    public DataflowTestData(Class<?> clazz) {
+        readStartLength = Arrays.asList(KV.of(100, 50), KV.of(140, 100), KV.of(100000, 10), KV.of(299999, 10));
+
+        // TODO Make reads construction more general.
+        reads = Lists.newArrayList(
+                makeRead(readStartLength.get(0), 1, clazz),
+                makeRead(readStartLength.get(1), 2, clazz),
+                makeRead(readStartLength.get(2), 3, clazz),
+                makeRead(readStartLength.get(3), 4, clazz));
+
+        kvRefShardiReads =  Arrays.asList(
+                KV.of(new ReferenceShard(0, "1"), Lists.newArrayList(reads.get(1), reads.get(0))),
```

Comment:
Do these values depend on the sharding size for correctness?  Could that be made explicit, maybe by making the start positions some fraction of the shard size constant?  It would be awkward if changing the shard size made the tests fail.